main.tf

```diff
@@ -11,6 +11,63 @@ resource "ibm_resource_instance" "scc_instance" {
   tags              = var.resource_tags
 }
 
+data "ibm_scc_provider_types" "scc_provider_types" {
+  count       = var.attach_wp_to_scc_instance ? 1 : 0
+  instance_id = ibm_resource_instance.scc_instance.guid
+}
+
+locals {
+  provider_types_list = var.attach_wp_to_scc_instance ? data.ibm_scc_provider_types.scc_provider_types[0].provider_types : []
+
+  provider_types_map = {
+    for provider_type in local.provider_types_list :
+    provider_type.name => provider_type
+  }
+
+  provider_type = lookup(local.provider_types_map, "workload-protection", null)
+}
+
+resource "ibm_scc_provider_type_instance" "scc_provider_type_instance_instance" {
+  depends_on       = [time_sleep.wait_for_authorization_policy]
+  count            = var.attach_wp_to_scc_instance ? 1 : 0
+  instance_id      = ibm_resource_instance.scc_instance.guid
+  attributes       = { "wp_crn" : var.wp_instance_crn }
+  name             = "workload-protection-instance"
+  provider_type_id = local.provider_type.id
+}
+
+# workaround for https://github.com/IBM-Cloud/terraform-provider-ibm/issues/4478
+resource "time_sleep" "wait_for_authorization_policy" {
+  depends_on = [ibm_iam_authorization_policy.scc_wp_s2s_access]
+
+  create_duration = "30s"
+}
+
+resource "ibm_iam_authorization_policy" "scc_wp_s2s_access" {
+  count                       = var.attach_wp_to_scc_instance ? 1 : 0
```

Comment:
you need to add the ability to skip this. Suggest adding a variable called `skip_scc_wp_auth_policy` and update the logic here include it